prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -105,10 +107,18 @@ class BlobAccess() : Logging {
          * @return the URL for the storage and container that this object represents
          */
         fun getBlobEndpoint(): String {
-            val match = blobEndpointRegex.find(connectionString)
-            val blobStorageUrl = match?.groups?.get("blobEndpoint")?.value
-                ?: throw RuntimeException("Connection string is misconfigured and does not contain a blob endpoint URL")
-            return "$blobStorageUrl/$containerName"
+            val blobEndpointMatch = blobEndpointRegex.find(connectionString)
+            val blobAccountNameMatch = blobAccountNameRegex.find(connectionString)
+            val blobEndpointSuffixMatch = blobEndpointSuffixRegex.find(connectionString)
+            val blobStorageUrl = blobEndpointMatch?.groups?.get("blobEndpoint")?.value
+            val blobAccountName = blobAccountNameMatch?.groups?.get("accountName")?.value
+            val blobEndpointSuffix = blobEndpointSuffixMatch?.groups?.get("endpointSuffix")?.value
+            if (blobStorageUrl != null) {
+                return "$blobStorageUrl/$containerName"
+            } else if (blobAccountName != null && blobEndpointSuffix != null) {
```

Comment:
I was not sure if including the endpoints in the source code was no good from a security front.  We could also consider following up and adding another ENV var that has this value, since it can be derived from the connection string it might not be worth it